src/common/src/hash/consistent_hash/vnode.rs

```diff
@@ -64,6 +64,9 @@ pub type AllVirtualNodeIter = std::iter::Map<std::ops::Range<usize>, fn(usize) -
 impl VirtualNode {
     /// The maximum value of the virtual node.
     pub const MAX: VirtualNode = VirtualNode::from_index(Self::COUNT - 1);
+    /// We may use `VirtualNode` as a datum in a stream, or store it as a column.
+    /// Hence this reifies it as a RW datatype.
+    pub const RW_TYPE: DataType = DataType::Int16;
```

Comment:
The `VirtualNode::MAX` value exceeds the maximum limit allowed by the `Int16` data type.

Reply:
I see, I will use i32 instead.

Reply:
VirtualNode::Max is 255:
```
Virtual node max: VirtualNode(255)
Virtual node max: VirtualNode(255)
Virtual node max: VirtualNode(255)
Virtual node max: VirtualNode(255)
```

Reply:
I saw `type VirtualNodeInner = u16;` which means the maximum value theoretically could be 65535. @BugenZhao Should we just use INT16 to store the vnode value?

Reply:
https://github.com/risingwavelabs/risingwave/blob/126782a4fd52a9a30afef59e1e066b7dfe5c46b4/src/frontend/src/expr/type_inference/func.rs#L619-L622

Seems like we do so else where too, for vnode derive from dist key in two phase agg.
Mentioning it in case we need to refactor all usage of vnode to u32 or some other datatype.
IMO we can do it in separate PR.

Reply:
Agree, we can do it in separate PR if necessary.

Reply:
Since we don't have an unsigned physical type, I guess using `I16` is okay here and we always interpret it as an unsigned number.

Reply:
> in case we need to refactor all usage of vnode to Int32 or some other datatype.

65535 storage partitions and computing parallelism should be sufficient for a streaming job, so it may not be necessary to exceed this limit. Additionally, the value can fit within an `i32` with overflowing. 😂